### PR TITLE
feat(YouTube): add `Hide Suggestions shelf` patch

### PR DESCRIPTION
**Pull request by johnconner122**
depends on:

 - https://github.com/ReVanced/revanced-patches/pull/2875

### CODE REVIEW

**Comment by oSumAtrIX** [comment]
What is a "suggestions shelf"?

**Comment by johnconner122** [comment]
> What is a "suggestions shelf"?

They are all similar to breaking news shelf.

**Comment by oSumAtrIX** [comment]
That does not tell me anything.

**Comment by johnconner122** [comment]
Watch it again


vT70Uwx_d

**Comment by johnconner122** [comment]
Breaking News


7aJQgyG_d

**Comment by johnconner122** [comment]
There are others like them as mentioned in: https://github.com/ReVanced/revanced-patches/issues/2779

**Comment by johnconner122** [comment]
They don't appear all the time, so it's harder to get a screenshot at the moment.

**Comment by oSumAtrIX** [comment]
Sounds good, I'll try to refactor this later.

**Comment by YT-Advanced** [comment]
If you can, I think you should seperate these shelf

File: app/src/main/java/app/revanced/integrations/utils/NavBarIndexHook.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    /**
+     * Check if current tab is Home Tab
+     */
+
+    public static boolean isHomeTab() {
+        return currentNavBarIndex == 0;
+    }
+
+    /**
+     * Check if current tab is Library Tab
```
Comment (YT-Advanced) [inline app/src/main/java/app/revanced/integrations/utils/NavBarIndexHook.java:37]: Wrong description here

File: app/src/main/java/app/revanced/integrations/patches/HideBreakingNewsPatch.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        return SettingsEnum.SPOOF_APP_VERSION.getBoolean()
+                && SettingsEnum.SPOOF_APP_VERSION_TARGET.getString().compareTo("17.30.35") < 0;
+    }
+
+    /**
+     * Injection point.
+     */
+    public static void hideBreakingNews(View view) {
+        if (!SettingsEnum.HIDE_SUGGESTIONS_SHELF.getBoolean()
+                || isSpoofingOldVersionWithHorizontalCardListWatchHistory()) return;
```
[the remark this answers is not in the document]
Comment (johnconner122) [reply app/src/main/java/app/revanced/integrations/patches/HideBreakingNewsPatch.java]: @LisoUseInAIKyrios Tablet mode seems to be using old breaking news code which needs bytecode patch, I think we should include it, we can just return if spoofing is enabled

**Comment by johnconner122** [comment]
Closing until a solution is found.

**Pull request closed without merging**